pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -303,12 +303,16 @@ func (f *Fit) isSchedulableAfterPodChange(logger klog.Logger, pod *v1.Pod, oldOb
 // isResourceScaleDown checks whether the resource request of the modified pod is less than the original pod
 // for the resources requested by the pod we are trying to schedule.
 func (f *Fit) isResourceScaleDown(targetPod, originalOtherPod, modifiedOtherPod *v1.Pod) bool {
-	if modifiedOtherPod.Spec.NodeName == "" {
-		// no resource is freed up whatever the pod is modified.
+	if !(modifiedOtherPod.Namespace == targetPod.Namespace && modifiedOtherPod.Name == targetPod.Name) &&
+		modifiedOtherPod.Spec.NodeName == "" {
+		// the other pod is not the target pod we are trying to schedule and it is not scheduled,
+		// so it doesn't free up any resource for the target pod.
 		return false
 	}
 
-	// the other pod was scheduled, so modification or deletion may free up some resources.
+	// the other pod was scheduled, modification or deletion may free up some resources.
+	// or the other pod is the target pod we are trying to schedule, modification may reduce
+	// the resource request of the target pod.
```

Comment:
We can just remove these comments since we'll describe at the top (after my suggestion applied).